vochain/state/process.go

```diff
@@ -211,9 +264,28 @@ func (v *State) UpdateProcess(p *models.Process, pid []byte) error {
 	if p == nil || len(p.ProcessId) != types.ProcessIDsize {
 		return ErrProcessNotFound
 	}
+	// update the process
 	v.Tx.Lock()
-	defer v.Tx.Unlock()
-	return updateProcess(&v.Tx, p, pid)
+	if err := updateProcess(&v.Tx, p, pid); err != nil {
+		return err
+	}
+	v.Tx.Unlock()
+	// try to update startBlocks database
```

Comment:
@p4u I dont know if this file is the best place for this code or if it needs to be moved to `transaction/transaction.go` because its not about the state at all.

Reply:
LGTM